File: skills/write-readable-code/assets/layout/jvm/template.java
package com.example.component;

import java.util.List;

public final class PublicType {
    public static final int PUBLIC_CONSTANT = 1;
    static final int PACKAGE_CONSTANT = 2;
    private static final int PRIVATE_CONSTANT = 3;

    public PublicType() {}

    public void publicMethod() {}

    void packageMethod() {}

    protected void protectedMethod() {}

    private void privateMethod() {}

    private static final class PrivateNestedType {}
}

// Prefer one top-level public type per file.
// Package-private top-level helpers may follow when justified.
final class PackagePrivateHelper {}

File: skills/write-readable-code/assets/layout/jvm/template.kt
package com.example.component

const val PUBLIC_CONSTANT = 1
internal const val MODULE_CONSTANT = 2
private const val PRIVATE_CONSTANT = 3

open class PublicType {
    fun publicMethod(): Int = privateMethod() + MODULE_CONSTANT

    internal fun moduleMethod(): Int = PUBLIC_CONSTANT

    protected open fun protectedMethod(): Int = PRIVATE_CONSTANT

    private fun privateMethod(): Int = protectedMethod()
}

internal class InternalType {
    fun run(): Int = PublicType().publicMethod()
}

private class FilePrivateType
